### PR TITLE
Fix mixin and validatable docs

### PR DESCRIPTION
**Pull request by ssh24**
Docs checklist:
 - [x] RelationMixin
 - [x] ObserverMixin
 - [x] TransactionMixin
 - [x] Validatable

connect to https://github.com/strongloop-internal/scrum-apex/issues/180

### CODE REVIEW

**Comment by rmg** [comment]
Node 0.10/0.12 have been removed from the config on the Windows CI. Next time the branch is rebased those statuses should go away.

File: lib/observer.js [hunk 1]
```diff
@@ -20,12 +20,28 @@ function ObserverMixin() {
 
 /**
  * Register an asynchronous observer for the given operation (event).
+ *
+ * Example:
+ *
```
Comment (b-admike) [inline lib/observer.js]: If you can add a description before the code, it'll help users understand what it does.

File: lib/observer.js [hunk 2]
```diff
@@ -65,9 +100,28 @@ ObserverMixin.clearObservers = function(operation) {
...
+    if (err) return cb(err);
+    // user can specify the logic after the observers have been notified
+ });
+ * ```
+ *
  * @param {String|String[]} operation The operation name(s).
  * @param {Object} context Operation-specific context.
- * @param {function(Error=)} callback The callback to call when all observers
+ * @callback {function(Error=)} callback The callback to call when all observers
  *   has finished.
```
Comment (b-admike) [inline lib/observer.js]: `when all observers have finished`

File: lib/observer.js [hunk 3]
```diff
@@ -123,19 +177,41 @@ ObserverMixin._notifyBaseObservers = function(operation, context, callback) {
 };
 
 /**
- * Run the given function with before/after observers. It's done in three serial
- * steps asynchronously:
+ * Run the given function with before/after observers.
+ *
+ * It's done in three serial steps asynchronously:
```
Comment (b-admike) [inline lib/observer.js]: nitpick: I suggest `serial asynchronous steps`

File: lib/observer.js (hunk 2 above)
Comment (kjdelisle) [inline lib/observer.js]: Oh, cool. Didn't know you could specify the type of the parameter it expects like that.

File: lib/observer.js (hunk 1 above)
Comment (ssh24) [inline lib/observer.js]: Fixed. The description is below.

File: lib/relations.js [hunk 4]
```diff
@@ -50,25 +50,33 @@ function RelationMixin() {
...
  *  });
  *
  *   // Query chapters for the book
- *   book.chapters(function(err, chapters) {  // all chapters with bookId = book.id
+ *   book.chapters(function(err, chapters) {
+ *     // all chapters with bookId = book.id
  *     console.log(chapters);
  *   });
  *
+ *   // Query chapters for the back with a filter
```
Comment (b-admike) [inline lib/relations.js]: `for the book`

File: lib/relations.js [hunk 5]
```diff
@@ -123,15 +131,20 @@ RelationMixin.hasMany = function hasMany(modelTo, params) {
  * @param {Class|String} modelTo Model object (or String name of model) to which you are creating the relationship.
  * @options {Object} params Configuration parameters; see below.
  * @property {String} as Name of the property in the referring model that corresponds to the foreign key field in the related model.
+ * @property {String} primaryKey Property name of primary key field.
  * @property {String} foreignKey Name of foreign key property.
- *
+ * @property {Object|Function} scope Explictly define additional scopes.
```
Comment (b-admike) [inline lib/relations.js]: Typo: `Explicitly`

File: lib/relations.js [hunk 6]
```diff
@@ -155,28 +168,432 @@ RelationMixin.belongsTo = function(modelTo, params) {
...
+ * Examples:
+ * ```
+ * Supplier.hasOne(Account, {as: 'account', foreignKey: 'supplierId'});
+ * ```
+ *
+ * If the target model doesn’t have a foreign key property, LoopBack will add a property with the same name.
+ *
+ * The type of the property will be the same as the type of the target model’s id property.
+ *
+ * Please note the foreign key property is defined on the target model (for example, Account).
```
Comment (b-admike) [inline lib/relations.js]: I suggest `in this example, Account`

File: lib/relations.js [hunk 7]
```diff
@@ -155,28 +168,432 @@ RelationMixin.belongsTo = function(modelTo, params) {
...
+ * Create a new account for the supplier. If there is already an account, an error will be reported.
+ * ```js
+ * supplier.account.create(data, function(err, account) {
+ *  ...
+ * });
+ * ```
+ *
+ * Find the supplier's account model.
+ * ```js
+ * supplier.account(function(err, account) {
```
Comment (b-admike) [inline lib/relations.js]: I reckon this is same as doing `supplier.account.find`?

Comment (ssh24) [inline lib/relations.js]: yes.

**Pull request merged**